nUpdate.Shared/Updating/UpdateManager.cs

```diff
@@ -472,5 +486,56 @@ bool Validate(KeyValuePair<UpdateVersion, string> filePathItem)
+        /// <param name="literalVersion">The update-packages literalVersion. <see cref="UpdateConfiguration.LiteralVersion"/></param>
+        /// <returns>
+        ///     Return converted URI.
+        ///     If <see cref="UpdateManager.UseDynamicUpdateUri"/> is false, return the original value.
+        /// </returns>
+        /// <exception cref="System.ArgumentNullException">updatePackageUri or literalVersion is null.</exception>
+        /// <exception cref="System.ArgumentException">literalVersion is a valid value. Reference <see cref="UpdateVersion"/></exception>
+        /// <exception cref="System.InvalidOperationException">updatePackageUri.AbsoluteUri is a valid uri.</exception>
+        private Uri ConvertPackageUri(Uri updatePackageUri, string literalVersion)
+        {
+            if (!UseDynamicUpdateUri)
+                return updatePackageUri;
+            if (updatePackageUri == null)
+                throw new ArgumentNullException(nameof(updatePackageUri));
+            if (literalVersion is null)
+                throw new ArgumentNullException(nameof(literalVersion));
+            if (!UpdateVersion.IsValid(literalVersion))
+                throw new ArgumentException($"{literalVersion} is not valid value.", nameof(literalVersion));
+            var packageName = updatePackageUri.AbsoluteUri.Substring(updatePackageUri.AbsoluteUri.LastIndexOf(@"/") + 1);
+            var updateConfigUrl = UpdateConfigurationFileUri.AbsoluteUri.Substring(0, UpdateConfigurationFileUri.AbsoluteUri.LastIndexOf(@"/"));
+            return new Uri($"{updateConfigUrl}/{literalVersion}/{packageName}");
+        }
+
+        /// <summary>
+        ///     Convert <paramref name="statisticsUri"/> to the one that corresponds to the directory
+        ///     where aslo the configuration file that is located in <see cref="UpdateManager.UpdateConfigurationFileUri"/>.
+        ///     If <see cref="UseDynamicUpdateUri"/> is false, return the original value.
+        /// </summary>
+        /// <param name="statisticsUri">The update-packages uri. <see cref="UpdateConfiguration.UpdatePhpFileUri"/></param>
+        /// <returns>
+        ///     Return converted URI.
+        ///     If <see cref="UpdateManager.UseDynamicUpdateUri"/> is false, return the original value.</returns>
+        /// <exception cref="System.ArgumentNullException">statisticsUri is null.</exception>
+        /// <exception cref="System.InvalidOperationException">statisticsUri.AbsoluteUri is a valid uri.</exception>
+        private Uri ConvertStatisticsUri(Uri statisticsUri)
+        {
+            if (!UseDynamicUpdateUri)
+                return statisticsUri;
+            if (statisticsUri == null)
+                throw new ArgumentNullException(nameof(statisticsUri));
+            var packageName = statisticsUri.AbsoluteUri.Substring(statisticsUri.AbsoluteUri.LastIndexOf(@"/") + 1);
```

Comment:
The statistics file does not have a package name. It always is a "statistics.php". So, you can safely remove this and just finish the string with "statistics.php". Update dependant data will be handled over using URI parameters later.

Reply:
I eventually chose to use `Uri.Segments.Last()` instead of "statistics.php" make it work after "statistics.php" was modified by someone else.